### PR TITLE
Fix broken translation status if only minor public versions exists

### PR DESCRIPTION
**Pull request by JoeyStk**
### Short description
Fixes the issue with the broken translation status if only minor public versions exists for POI and event

### Proposed changes
- Add extra if-condition to ensure the correct translation status is set

### Side effects
I don't think there are any.  

### Resolved issues
Fixes: #1788

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/cms/views/events/event_form_view.py [hunk 1]
```diff
@@ -236,6 +236,13 @@ def post(self, request, **kwargs):
                     language__in=languages
                 ).update(status=status.DRAFT)
 
+            if (
+                event_translation_form.instance.status == status.PUBLIC
+                and event_translation_form.instance.minor_edit
+            ):
+                event_translation_form.instance.poi.translations.filter(
```
Comment (MizukiTemma) [inline integreat_cms/cms/views/events/event_form_view.py:243]:
```suggestion
                event_translation_form.instance.event.translations.filter(
```

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
Thank you for PR 👍 

I think this is just a copy & paste thing though have to suggest a change 😅 See code suggestion.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ UNRELEASED
 * [ [#1521](https://github.com/digitalfabrik/integreat-cms/issues/1521) ] Show warning when user entered slug got changed
 * [ [#1807](https://github.com/digitalfabrik/integreat-cms/issues/1807) ] Remove short description field from POI form
 * [ [#522](https://github.com/digitalfabrik/integreat-cms/issues/522) ] Add region setting to activate SEO section, add SEO section to POI form
+* [ [#1788](https://github.com/digitalfabrik/integreat-cms/issues/1788) ] Fix broken translation status if only minor public versions exists
```
Comment (timoludwig) [inline CHANGELOG.md:8]:
```suggestion
* [ [#1788](https://github.com/digitalfabrik/integreat-cms/issues/1788) ] Fix broken translation status of events & locations if only minor public versions exists
```

File: integreat_cms/cms/views/events/event_form_view.py [hunk 3]
```diff
@@ -236,6 +236,13 @@ def post(self, request, **kwargs):
                     language__in=languages
                 ).update(status=status.DRAFT)
 
+            if (
```
Comment (timoludwig) [inline integreat_cms/cms/views/events/event_form_view.py:239]:
```suggestion
            elif (
```
We don't need to check this condition when the status is `DRAFT`

File: integreat_cms/cms/views/pois/poi_form_view.py [hunk 4]
```diff
@@ -177,6 +177,13 @@ def post(self, request, *args, **kwargs):
                 poi_translation_form.instance.poi.translations.filter(
                     language__in=languages
                 ).update(status=status.DRAFT)
+            if (
```
Comment (timoludwig) [inline integreat_cms/cms/views/pois/poi_form_view.py:180]:
```suggestion
            elif (
```

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks a lot, works as expected! :+1:

File: integreat_cms/cms/views/events/event_form_view.py (hunk 1 above)
Comment (timoludwig) [reply integreat_cms/cms/views/events/event_form_view.py:243]: It's a bit sad that our tests did not reveal this error. Unfortunately we can't add automated tests for every edge case of the form submissions... at least for the future, we could test this specific case by adding
```
            (
                "edit_event",
                PRIV_STAFF_ROLES + REGION_ROLES,
                {
                    "title": "new title",
                    "start_date": "2030-01-01",
                    "end_date": "2030-01-01",
                    "is_all_day": True,
                    "status": status.PUBLIC,
                    "minor_edit": True
                },
            ),
```
to the [view test config](https://github.com/digitalfabrik/integreat-cms/blob/3ed0649012dfa9945ea183e35ae401ef0a950413/tests/cms/views/view_config.py#L736-L746).

**Comment by JoeyStk** [comment]
Thank you :) All of it makes sense :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,7 @@ UNRELEASED
 * [ [#522](https://github.com/digitalfabrik/integreat-cms/issues/522) ] Add region setting to activate SEO section, add SEO section to POI form
 * [ [#1858](https://github.com/digitalfabrik/integreat-cms/issues/1858) ] Enable submitting feedback about fallback translations of recurring events
 * [ [#1865](https://github.com/digitalfabrik/integreat-cms/issues/1865) ] Fix TextLab HIX widget for non-staff users
+* [ [#1788](https://github.com/digitalfabrik/integreat-cms/issues/1788) ] Fix broken translation status of events & locations if only minor public versions exists
```
Comment (timoludwig) [inline CHANGELOG.md:15]: Could you put this entry into the "unreleased" section?

**Review by timoludwig — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (JoeyStk) [reply CHANGELOG.md]: Oh shit, yes of course

**Review by timoludwig — APPROVED** [review]

**Review by MizukiTemma — APPROVED** [review]

**Comment by timoludwig** [comment]
> When DRAFT event/POI is published, translation status is correct now ✅
> 
> However when status is still DRAFT, it shows "translation missing". Is it an expected behavior or another issue?

This was defined as expected behavior by the service team since all draft content is not visible in the app and is thus "missing".

**Comment by seluianova** [comment]
@timoludwig looks a bit strange to me, because "missing" is more like "not exists". And to mark thing as "visible" there is a "published" status... But okay, thanks for the answer :)

**Comment by timoludwig** [comment]
@seluianova yes, I agree, this probably needs to be revisited in the future together with the service team and UI/UX.
